tmc-langs-util/src/main/java/fi/helsinki/cs/tmc/langs/util/tarservice/TarCreator.java

```diff
@@ -0,0 +1,81 @@
+package fi.helsinki.cs.tmc.langs.util.tarservice;
+
+import static org.apache.commons.compress.archivers.tar.TarArchiveOutputStream.LONGFILE_POSIX;
+
+import org.apache.commons.compress.archivers.ArchiveException;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveOutputStream;
+import org.apache.commons.compress.utils.IOUtils;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public class TarCreator {
+
+    /**
+     * Copies the location of tmc-langs-cli and tmcrun to unzipped project folder
+     * and creates a tarball.
+     *
+     * @param projectDir     Location of unzipped project dir
+     * @param tmcLangs       Location of tmc-langs-cli.jar
+     * @param tmcrun         Location of tmc-run init script
+     * @param targetLocation Location where the tar archive should be extracted to
+     * @throws IOException      Error!
+     * @throws ArchiveException Error!
+     */
+    public void createTarFromProject(Path projectDir, Path tmcLangs, Path tmcrun,
+                                     Path targetLocation) throws IOException, ArchiveException {
+        Files.copy(tmcrun, projectDir.resolve(tmcrun.getFileName()));
+        Files.copy(tmcLangs, projectDir.resolve(tmcLangs.getFileName()));
+        createTarBall(projectDir, targetLocation);
+    }
+
+    /**
+     * Creates a tarball from a directory.
+     *
+     * @param project        Project directory file
+     * @param targetLocation Location where the tar archive should be extracted to
+     * @throws IOException      Error!
+     * @throws ArchiveException Error!
+     */
+    private void createTarBall(Path project, Path targetLocation)
+            throws IOException, ArchiveException {
+        try (FileOutputStream tarOut = new FileOutputStream(targetLocation.toString())) {
+            try (TarArchiveOutputStream aos = new TarArchiveOutputStream(tarOut)) {
```

Comment:
**Comment, not a request for changes:** You could combine these two `try`lines into a single line as 
```
try (TarArchiveOutputStream aos = new TarArchiveOutputStream(new FileOutputStream(targetLocation.toString()))) {
```

The `try`-clause then calls `close()` for `aos`, which then calls `close()` of the `FileOutputStream` explicitly. But that's getting a bit wordy and I'm not sure I like it.

Alternatively you could have multiple statements in a single try-with-resources statement, like this:
```
try (FileOutputStream tarOut = new FileOutputStream(targetLocation.toString());
      TarArchiveOutputStream aos = new TarArchiveOutputStream(tarOut)) {
```

But that's also kinda ugly and also ends up calling `close`on `tarOut` twice (once implicitly thru `aos`, once explicitly in the (hidden) `finally`block).

**The current implementation is fine in my books**, as I find all three possible solutions sort-of-verbose.  

If you are interested in these kinds of things, there's some discussion on a similar case here: https://stackoverflow.com/questions/12552863/correct-idiom-for-managing-multiple-chained-resources-in-try-with-resources-bloc